test/e2e/networking.go

```diff
@@ -102,160 +103,245 @@ var _ = Describe("Networking", func() {
+
+			// Clean up service
+			defer func() {
+				By("Cleaning up the service")
+				if err := f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
+					Failf("unable to delete svc %v: %v", svc.Name, err)
 				}
-			case strings.Contains(string(body), "no endpoints available"):
-				Logf("Attempt %v: waiting on service/endpoints", i)
-			default:
-				Logf("Unexpected response:\n%s", body)
-			}
-		}
+			}()
+
+			Logf("launching pod per node.....")
+			podNames := launchNetTestPodPerNode(port, nettestVersion, f, nodes, svcname)
+			// Clean up the pods
+			defer func() {
+				By("Cleaning up the webserver pods")
+				for _, podName := range podNames {
+					if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
+						Logf("Failed to delete pod %s: %v", podName, err)
+					}
+				}
+			}()
+			Logf("Launched test pods for %v", port)
+			By("waiting for all of the webserver pods to transition to Running + reaching the Passing state.")
 
-		if !passed {
-			if body, err = getDetails(); err != nil {
-				Failf("Timed out. Cleaning up. Error reading details: %v", err)
-			} else {
-				Failf("Timed out. Cleaning up. Details:\n%s", string(body))
+			for _, podName := range podNames {
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred())
+				By(fmt.Sprintf("waiting for connectivity to be verified [ pod = %v ,  port =  %v ] ", podName, port))
+				//once response OK, evaluate response body for pass/fail.
+				passedPolling := pollPeerStatus(f, svc, timeoutSeconds)
+				if !passedPolling {
+					Failf("Failed at polling %v %v", podName, port)
+				}
 			}
-		}
-		Expect(string(body)).To(Equal("pass"))
-	})
+			Logf("Finished test pods for %v", port)
 
-})
+			//Once the polling for a service is completed,
+			//we send failure/success to the completion channel.
+			completionChannel <- svcname
+		} //testP
+
+		//Test all nodes over port p.
+		go testP(nodes, ports[p], portCompleteChannel)
+
+	} //for
+
+	//For loop that increments every time a nettest over one of the ports completes.
+	for pReturned := range ports {
+		Logf("Waiting on ports to report back.  So far %v have been discovered...", pReturned)
+		Logf("... Another service has successfully been discovered: %v ( %v ) ", pReturned, <-portCompleteChannel)
+	}
+	Logf("Completed test on %v port/services", len(ports))
+}
+
+//makePorts makes a bunch of ports from 8080->8080+n
+func makePorts(n int) []int {
+	m := make([]int, n)
+	for i := 0; i < n; i++ {
+		m[i] = 8080 + i
+	}
+	return m
+}
 
-func LaunchNetTestPodPerNode(f *Framework, nodes *api.NodeList, name, version string) []string {
+//launchNetTestPodPerNode launches nettest pods, and returns their names.
+func launchNetTestPodPerNode(port int, version string, f *Framework, nodes *api.NodeList, name string) []string {
 	podNames := []string{}
 
+	Logf("launching net test pod per node %v ", len(nodes.Items))
```

Comment:
it's not clear that from the log message the number printed is supposed to be a length.